web/components/modals/ChildModal.tsx

```diff
@@ -1,28 +1,62 @@
-import React from "react";
+import SearchableDropdown from "@components/atoms/SearchInput";
+import { getCaregivers } from "db/actions/admin/Caregiver";
+import React, { useEffect, useState } from "react";
 import { useForm } from "react-hook-form";
 import { BiLoaderCircle } from "react-icons/bi";
 import { FaTimes } from "react-icons/fa";
 
+interface CaregiverOption {
+  label: string;
+  value: any;
+}
+
 function ChildModal({
   setModal,
   onSubmit,
-  caretakers,
-  buttonText = "Add a Child",
+  buttonText = "Add child",
   header = "Add a Child",
   values,
 }: {
   setModal: (modal: boolean) => void;
   onSubmit: (data: any) => void;
-  caretakers: { name: string; id: string }[];
   buttonText?: string;
   header?: string;
   values?: any;
 }) {
   const {
     register,
+    setValue,
     handleSubmit,
     formState: { errors, isSubmitting },
   } = useForm();
+  const [caregivers, setCaregivers] = useState<any[]>([]);
+
+  const handleCaretakerChange = (selectedCaretaker: any) => {
+    if (selectedCaretaker) {
+      setValue("caretakerName", selectedCaretaker.label); 
+      setValue("caretakerID", selectedCaretaker.value.id);
+      setValue("caretaker", selectedCaretaker.value); 
+    }
+  }
+
+  const fetchCaregivers = async () => {
+    try {
+        const fetchedData = await getCaregivers();
+        setCaregivers(fetchedData.map((caregiver) => (
+          { label: caregiver.name, value: caregiver }
+        )));
+    } catch (error) {
+        console.error('Error fetching records.');
```

Comment:
Return an indication of the error like an object that specifies the error or throwing an error so that we can deal with this (admittedly rare) issue in a later sprint!